testsuite/tests/libratest/smoke_test.rs

```diff
@@ -283,15 +283,19 @@ fn smoke_test_single_node() {
 
 #[test]
 fn smoke_test_single_node_block_metadata() {
-    let (_swarm, mut client_proxy) = setup_swarm_and_client_proxy(1, 0);
+    let (swarm, mut client_proxy) = setup_swarm_and_client_proxy(1, 0);
     // just need an address to get the latest version
     let address = AccountAddress::from_hex_literal("0xA550C18").unwrap();
-    // sleep 1s to commit some blocks
-    thread::sleep(time::Duration::from_secs(1));
```

Comment:
we should ban folks from introducing sleep methods into tests...

Reply:
Agreed, but I'd go one further, ban all threaded logic and have a single, configurable threading solution. 

Maybe pub/sub message passing solution (events).   Or an actor model.  With cycle prevention by analyzing the data flow.   

It's not enough to say what not do, but also must provide a simple, complete, ergonomic solution otherwise we'll just push people to worse forms of temporal coupling (distributed through more files/crates/less obviously "bad").